packages/api/tests/models/granules/test-granule-reporting.js

```diff
@@ -253,6 +267,96 @@ test('_storeGranuleRecord() will allow a running status to replace a failed stat
   );
 });
 
+test('_storeGranuleRecord() will allow a completed status to replace a queued status for a new execution', async (t) => {
+  const { granuleModel } = t.context;
+
+  const granule = fakeGranuleFactoryV2({ status: 'queued' });
+
+  await granuleModel._storeGranuleRecord(granule);
+  const updateTime = Date.now();
+
+  const updatedGranule = {
+    ...granule,
+    execution: 'new-execution-url',
+    status: 'completed',
+    createdAt: updateTime,
+  };
+
+  await granuleModel._storeGranuleRecord(updatedGranule);
+
+  const fetchedItem = await granuleModel.get({ granuleId: granule.granuleId });
+
+  t.deepEqual(
+    fetchedItem,
+    {
+      ...granule,
+      status: 'completed',
+      createdAt: updateTime,
+      execution: 'new-execution-url',
+    }
+  );
+});
+
+test('_storeGranuleRecord() will allow a running status to replace a queued status for a new execution', async (t) => {
```

Comment:
This test case isn't bad to have, but I think we at least want another unit test to prove that a running status can replace a queued status for the same execution, since that is the typically expected case